Mathlib/RingTheory/Etale/Field.lean

```diff
@@ -148,4 +153,61 @@ theorem iff_isSeparable [EssFiniteType K L] :
     FormallyEtale K L ↔ Algebra.IsSeparable K L :=
   ⟨fun _ ↦ FormallyUnramified.isSeparable K L, fun _ ↦ of_isSeparable K L⟩
 
+attribute [local instance]
+  IsArtinianRing.subtype_isMaximal_finite IsArtinianRing.fieldOfSubtypeIsMaximal in
+/--
+If `A` is an essentially of finite type algebra over a field `K`, then `A` is formally etale
+over `K` if and only if `A` is a finite product of separable field extensions.
+-/
+theorem iff_exists_algEquiv_prod [EssFiniteType K A] :
+    FormallyEtale K A ↔
+      ∃ (I : Type u) (_ : Finite I) (Ai : I → Type u) (_ : ∀ i, Field (Ai i))
+        (_ : ∀ i, Algebra K (Ai i)) (_ : A ≃ₐ[K] Π i, Ai i),
+        ∀ i, Algebra.IsSeparable K (Ai i) := by
+  classical
+  constructor
+  · intro H
+    have := FormallyUnramified.finite_of_free K A
+    have := FormallyUnramified.isReduced_of_field K A
+    have : IsArtinianRing A := isArtinian_of_tower K inferInstance
+    letI : Fintype {I : Ideal A | I.IsMaximal} := (nonempty_fintype _).some
```

Comment:
The imports of `MaximalSpectrum` is pretty light now, and you can feel free to import it. I need to change the file Artinian to use it though ...

Reply:
I take it back; it's still heavy enough that you can't import it freely without acquiring the large-import label ...